ui/src/BaseEditor.ts

```diff
@@ -0,0 +1,146 @@
+    public abstract getSaveConfirmationText(context: ContextT): Promise<string>;
+
+    public async showEditor(context: ContextT, sizeLimit?: number /* in Megabytes */): Promise<void> {
+        const fileName: string = await this.getFilename(context);
+
+        this.appendToOutput(localize('opening', 'Opening "{0}" ...', fileName));
+        if (sizeLimit) {
+            const size: number = await this.getSize(context);
+            if (size > sizeLimit) {
+                const message: string = localize('tooLargeError', '"{0}" is too large to download.', fileName);
+                this.appendLineToOutput(localize('failed', " Failed."));
+                throw new Error(message);
+            }
+        }  else {
+            try {
+                const localFilePath: string = await TemporaryFile.create(fileName);
+                const document: vscode.TextDocument = await vscode.workspace.openTextDocument(localFilePath);
+                this.fileMap[localFilePath] = [document, context];
+                const data: string = await this.getData(context);
+                const textEditor: vscode.TextEditor = await vscode.window.showTextDocument(document);
+                await this.updateEditor(data, textEditor);
+                this.appendLineToOutput(' Done.');
+            } catch (error) {
+                let message: string;
+                this.appendLineToOutput(localize('failed', " Failed."));
+                // tslint:disable-next-line:no-unsafe-any
+                if (!!error.message) {
+                    // tslint:disable-next-line:no-unsafe-any
+                    message = error.message;
+                } else {
+                    message = JSON.stringify(error);
+                }
+                throw new Error(message);
+            }
+        }
+    }
+
+    public async updateMatchingContext(doc: vscode.Uri): Promise<void> {
+        const filePath: string | undefined = Object.keys(this.fileMap).find((fsPath: string) => path.relative(doc.fsPath, fsPath) === '');
+        if (filePath) {
+            const [textDocument, context]: [vscode.TextDocument, ContextT] = this.fileMap[filePath];
+            await this.updateRemote(context, textDocument);
+        }
+    }
+
+    public async dispose(): Promise<void> {
+        // tslint:disable-next-line:no-unsafe-any
+        Object.keys(this.fileMap).forEach(async (key: string) => await fse.remove(path.dirname(key)));
+    }
+
+    public async onDidSaveTextDocument(globalState: vscode.Memento, doc: vscode.TextDocument): Promise<void> {
+        const filePath: string | undefined = Object.keys(this.fileMap).find((fsPath: string) => path.relative(doc.uri.fsPath, fsPath) === '');
+        if (!this.ignoreSave && filePath) {
+            const context: ContextT = this.fileMap[filePath][1];
+            const showSaveWarning: boolean | undefined = vscode.workspace.getConfiguration().get(this.showSavePromptKey);
+            let shouldUpdateRemote: boolean = false;

```

Comment:
Remove the `shouldUpdateRemote` variable. (Jake has removed it in his code, too. He only had it initially because he wasn't throwing a `UserCancelledError`. That's the only place where `shouldUpdateRemote` is not set to true and it doesn't matter because the error is thrown)